pkg/commands/compute/compute_publish_test.go

```diff
@@ -16,242 +15,137 @@ import (
 )
 
 func TestPublish(t *testing.T) {
+	// We're going to chdir to a temp environment,
```

Comment:
The changes to this function can be summarised as follows...

1. Deduplicate a lot of the test configuration.
2. Define the test configuration outside of the test harness.
3. Define a single temp environment rather than having a new environment for each test scenario.